cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2021
+ * United States Army Corps of Engineers - Hydrologic Engineering Center (USACE/HEC)
+ * All Rights Reserved.  USACE PROPRIETARY/CONFIDENTIAL.
+ * Source may not be released without written approval from HEC
+ */
+
+package mil.army.usace.hec.cwms.radar.client.controllers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import mil.army.usace.hec.cwms.radar.client.ClientNotFoundException;
+import mil.army.usace.hec.cwms.radar.client.NoDataFoundException;
+import mil.army.usace.hec.cwms.radar.client.model.TimeSeries;
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.Test;
+
+class TestTimeSeriesController {
+
+    private static final String BASE_URL = "http://localhost:11524";
+
+    @Test
+    void testRetrieveTimeSeries() throws IOException {
+        Path path = new File(getClass().getClassLoader().getResource("radar/json/timeseries.json")
+            .getFile()).toPath();
+        String collect = String.join("\n", Files.readAllLines(path));
+        MockWebServer server = new MockWebServer();
+        server.enqueue(new MockResponse().setBody(collect));
+        server.start();
+        Instant start = ZonedDateTime.of(2018, 1, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        Instant end = ZonedDateTime.of(2018, 2, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        TimeSeries timeSeries = new TimeSeriesController().retrieveTimeSeries(server::url,
+            "arbu.Elev.Inst.1Hour.0.Ccp-Rev", "SWT", "SI", "NAVD88",
+            start, end, null);
+        assertEquals(500, timeSeries.getValues().size());
+        assertEquals(745, timeSeries.getTotal());
+        assertEquals("SWT", timeSeries.getOfficeId());
+        assertEquals("m", timeSeries.getUnits());
+        assertEquals(Duration.ZERO, timeSeries.getInterval());
+        assertEquals("ARBU.Elev.Inst.1Hour.0.Ccp-Rev", timeSeries.getName());
+        assertEquals(start, timeSeries.getBegin().toInstant());
+        Instant lastTime = Instant.ofEpochMilli(
+            timeSeries.getValues().get(timeSeries.getValues().size() - 1).getDateTime());
+        assertTrue(end.isAfter(lastTime));
+    }
+
+    @Test
+    void testRetrieveTimeSeriesPagination() throws IOException {
```

Comment:
Add in a multi-threaded test?

Reply:
The test isn't for the server, its to ensure that our code can operate in that manner. Note that there is one object that is shared or should be shared, the OkHttpClient itself. See comment on HttpRequestBuilder about sharing the client per documentation.

Reply:
test one: MockHttpServer requires synchronous queueing and retrieval of responses. So I can't reuse the same mock server
test two: create new MockHttpServer for every thread. This throws a socket read time out even when only performing two threaded calls.

Reply:
scratch the, I was able to get the test to work, I was just using the wrong port

Reply:
OK, submitted mult-threaded test